wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/text/TextStyles.kt

```diff
@@ -0,0 +1,490 @@
+        public val DARK_RED: Color = Color(0xAA0000)
+        /** Minecraft's native dark purple color. */
+        public val DARK_PURPLE: Color = Color(0xAA00AA)
+        /** Minecraft's native gold color. */
+        public val GOLD: Color = Color(0xFFAA00)
+        /** Minecraft's native grey color. */
+        public val GREY: Color = Color(0xAAAAAA)
+        /** Minecraft's native dark grey color. */
+        public val DARK_GREY: Color = Color(0x555555)
+        /** Minecraft's native blue color. */
+        public val BLUE: Color = Color(0x5555FF)
+        /** Minecraft's native green color. */
+        public val GREEN: Color = Color(0x55FF55)
+        /** Minecraft's native aqua color. */
+        public val AQUA: Color = Color(0x55FFFF)
+        /** Minecraft's native red color. */
+        public val RED: Color = Color(0xFF5555)
+        /** Minecraft's native light purple color. */
+        public val LIGHT_PURPLE: Color = Color(0xFF55FF)
+        /** Minecraft's native yellow color. */
+        public val YELLOW: Color = Color(0xFFFF55)
+        /** Minecraft's native white color. */
+        public val WHITE: Color = Color(0xFFFFFF)
+
+        /**
+         * Gets a color from [TextColor] and converts it to [Color].
+         *
+         * @param color The [TextColor] to convert
+         * @return A [Color] from [color]
+         *
+         * @author NoComment1105
+         */
+        public fun fromTextColor(color: TextColor): Color {
+            return Color(color.rgb)
+        }
+
+        /**
+         * Gets a color from [Formatting] and converts it to [Color].
+         * If the color value is null, it defaults to black.
+         *
+         * @param formattingColor The color to convert
+         * @return A [Color] from [formattingColor] or black if invalid/null
+         *
+         * @author NoComment1105
+         */
+        public fun fromFormatting(formattingColor: Formatting): Color {
+            return Color(formattingColor.colorValue?.let(::Color)?.value ?: BLACK.value)
+        }
+    }
+
+    /**
+     * Converts [value] to a [TextColor].
+     *
+     * @return A [TextColor] created from the [value] value
+     *
+     * @author NoComment1105
+     */
+    public fun toTextColor(): TextColor {
+        return TextColor.fromRgb(value)
+    }
+}
+
+/**
+ * The builder to create a hover event for an item.
+ *
+ * @author NoComment1105
+ */
+@TextDslMarker
+public class ItemHoverEvent {
+    /** The item stack to apply the event to. */
+    public var itemStack: ItemStack? = null
+    /** The item to apply the event to. */
+    public var item: Item? = null
+    /** The NBT of the item to apply the event to. */
+    public var nbt: NbtCompound? = null
+
+    /**
+     * Gets the item based off the [itemStack], [item] or [nbt] variables.
+     *
+     * @author NoComment1105
+     */
+    private fun getItem(): ItemStack {
+        return if (itemStack != null) {
```

Comment:
maybe a `when`?

Reply:
Don't believe this is possible as different things are being checked against in each branch